app/routes/circulars/circulars.server.ts

```diff
@@ -107,8 +107,68 @@ export async function search({
   totalItems: number
 }> {
   const client = await getSearch()
+  const isLuceneSyntax = query ? query.includes(':') : false
```

Comment:
That's not sufficient to detect Lucene syntax. Even without colons, the query string may contain boolean operators (`AND`, `OR`) or parentheses.

I suggest that for simplicity and a consistent user experience, Lucene query syntax is either always on or always off. I suggest always on, because it's a nice feature to have!

Reply:
ok, added a regex that gets all these conditions:
: 
There should be text on either side of the :
Example:
Subject: something

AND
There should be text on either side of the AND
Example:
Dogs AND cats

OR
There should be text on either side of the OR
Example:
Dogs OR cats

?
This can occur anywhere in the query
Examples:
Thi?s
This?
?this

*
This can occur anywhere in the query
Examples:
thi*
*this
th*s

~
This symbol will have text before it
Example:
Term~

[TO]
This will have [ then some text and then TO and then more text and then ]
Example:
Date: [ some string TO some string]

+
This will have text following it
Example:
+dogs

NOT
This will have text before and after NOT
Example:
Dogs NOT cats

-
This will have text following it
Examples:
Dogs -cats
-cats bowls

()
This will have text inside the parens
Example:
(Something will be in here)


^, which can be supported as part of lucene syntax is taken care of with our weighting. This is not a supported search character.

Reply:
As I said, I suggest that we either have Lucene syntax always on or always off.

Reply:
I'm now convinced that it _does_ make sense to preprocess the query string. However, I think it has to be a bit more subtle:

- The query string may consist of multiple words (such as `x-shooter GRB follow-up`). If so, then we need to correctly insert wildcards, fuzzy modifiers, etc. for each term.
- If a term consists _only_ of digits, then we don't want to fuzz it.

I think that it will help to use one of the several [Lucene query syntax parser/formatters on NPM](https://www.npmjs.com/search?q=lucene) to tokenize the query string.